src/IdentityServer4/Validation/TokenRequestValidator.cs

```diff
@@ -573,21 +573,31 @@ public TokenRequestValidator(IdentityServerOptions options, IAuthorizationCodeSt
             return Valid(result.CustomResponse);
         }
 
-        private async Task<bool> ValidateRequestedScopesAsync(NameValueCollection parameters)
+        private async Task<bool> ValidateRequestedScopesAsync(NameValueCollection parameters, bool ignoreImplicitIdentityScopes = false, bool ignoreImplicitOfflineAccess = false)
         {
+            var ignoreIdentityScopes = false;
+            
             var scopes = parameters.Get(OidcConstants.TokenRequest.Scope);
-
             if (scopes.IsMissing())
             {
+                if (ignoreImplicitIdentityScopes)
+                {
+                    ignoreIdentityScopes = true;
+                }
+                
                 _logger.LogTrace("Client provided no scopes - checking allowed scopes list");
 
                 if (!_validatedRequest.Client.AllowedScopes.IsNullOrEmpty())
                 {
                     var clientAllowedScopes = new List<string>(_validatedRequest.Client.AllowedScopes);
-                    if (_validatedRequest.Client.AllowOfflineAccess)
+                    if (!ignoreImplicitOfflineAccess)
                     {
-                        clientAllowedScopes.Add(IdentityServerConstants.StandardScopes.OfflineAccess);
+                        if (_validatedRequest.Client.AllowOfflineAccess)
```

Comment:
You are sure that you dont want to log that you are removing them?

If a developer added them without knowing that they are not valid they may wonder why they are not there and there is no error.

Reply:
feel free to do a PR - log at debug level

Reply:
PR?

Reply:
Probably not I have some tasks at work that need to be completed before I go on vacation.    As its your code you probably know exact where to put it anyway.